Common/Securities/CashBuyingPowerModel.cs

```diff
@@ -81,6 +82,11 @@ public override HasSufficientBuyingPowerForOrderResult HasSufficientBuyingPowerF
                 orderQuantity = parameters.Order.AbsoluteQuantity;
             }
 
+            var amountPrecision = parameters.Security.SymbolProperties.LotSize.GetDecimalPlaces();
+
+            totalQuantity = Math.Round(totalQuantity, amountPrecision);
+            orderQuantity = Math.Round(orderQuantity, amountPrecision);
```

Comment:
Interesting, _I think_ this might be because the provided argument in `HasSufficientBuyingPowerForOrderParameters` aren't respecting the lot size

Reply:
I have code that does round to lot size before calling the HasSufficientBuyingPowerForOrderParameters. I can add some tests but that will be some time on weekends I believe

In case it's a buy, we multiply the values (parameters.Order.AbsoluteQuantity * GetOrderPrice(parameters.Security, parameters.Order)) which could cause rounding issues. 

Imagine these values: 20.5 * 2.01  - they produce 41.2005 which has the precision of 4, while the parameters have the precision of 1 and 2. So it's known that values with lesser precision can result in values with greater precision after division or multiplication has been performed on them. 

That's why we need to round order quantity which is quantity converted into quote currency. And after that conversion, the precision could become greater than the lot size.

The reason why we need to round total quantity - that's because it's not always valid in the context of lot size, for example after an order has been filled we make calculations in the ProcessFill method, but these calculations do not respect the lot size, again the mul/div operations can cause values of greater precision, which is not taken into consideration in the ProcessFill method. But now I am thinking that we shouldn't really round total quantity but perhaps better fix the error in the ProcessFill methods